src/logic/Utils.ts

```diff
@@ -63,34 +63,54 @@ export const parseIni = (data: string) => {
   };
   const value = {};
   const lines = data.split(/[\r\n]+/);
-  let section = null;
+  let section: string | null = null;
   lines.forEach(function(line) {
     if (regex.comment.test(line)) {
       return;
     } else if (regex.param.test(line)) {
-      const match = line.match(regex.param);
+      const match: string[] | null = line.match(regex.param);
 
       // TODO: github copilot made this part, but I have no idea what it does
       // if (match?.length !== 3) {
       //   throw "Could not parse INI file.";
       // }
 
       if (section) {
-        value[section][match[1]] = match[2];
+        value[section][match?.[1]] = match?.[2];
       } else {
-        value[match[1]] = match[2];
+        value[match![1]] = match?.[2];
       }
     } else if (regex.section.test(line)) {
       const match = line.match(regex.section);
-      value[match[1]] = {};
-      section = match[1];
+      value[match![1]] = {};
+      section = match![1];
     } else if (line.length == 0 && section) {
       section = null;
     }
   });
   return value;
 };
 
+/* Pretty much just reverse the above function */
+export const unparseIni = (data: SchemeIni) => {
```

Comment:
Maybe name this "generateIni" or something?

Reply:
@theRealPadster It's not really generating an ini, it's just reversing the above function. If the above function is parsing it and you undo the above function that would be unparsing it right?

Reply:
Is it not effectively taking the parsed object and generating the text for the ini file from it?